bulk_handlers_test.go

```diff
@@ -40,3 +41,185 @@ func TestBulkCreateMissingSourceType(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestCreateUserWithResourceOwnershipApplicationType(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	conf.ResourceOwnership = "user"
+
+	testUserId := "testUser"
+	identityHeader := testutils.IdentityHeaderForUser(testUserId)
+
+	nameSource := "test source"
+	sourceTypeName := "bitbucket"
+	applicationTypeName := "app-studio"
+	authenticationResourceType := "application"
+
+	requestBody := testutils.SingleResourceBulkCreateRequest(nameSource, sourceTypeName, applicationTypeName, authenticationResourceType)
+
+	body, err := json.Marshal(requestBody)
+	if err != nil {
+		t.Error("Could not marshal JSON")
+	}
+
+	c, _ := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/bulk_create",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+	c.Set("identity", &identity.XRHID{Identity: identityHeader})
+
+	err = BulkCreate(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var users []m.User
+	err = dao.DB.Model(&m.User{}).Find(&users).Error
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(users) != 1 {
+		t.Errorf("1 user expected instead of %d", len(users))
+	}
+
+	if users[0].UserID != testUserId {
+		t.Errorf("expected userid is %s instead of %s", testUserId, users[0].UserID)
+	}
+
+	err = dao.DB.Model(&m.User{}).Delete(&users).Error
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = cleanSourceForTenant(nameSource, &fixtures.TestTenantData[0].Id)
+	if err != nil {
+		t.Errorf(`unexpected error received when deleting the source: %s`, err)
+	}
+}
+
+func TestCreateUserWithoutResourceOwnershipApplicationType(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	conf.ResourceOwnership = "user"
+
+	testUserId := "testUser"
+	identityHeader := testutils.IdentityHeaderForUser(testUserId)
+
+	nameSource := "test source"
+	sourceTypeName := "amazon"
+	applicationTypeName := "cost-management"
+	authenticationResourceType := "application"
+
+	requestBody := testutils.SingleResourceBulkCreateRequest(nameSource, sourceTypeName, applicationTypeName, authenticationResourceType)
+
+	body, err := json.Marshal(requestBody)
+	if err != nil {
+		t.Error("Could not marshal JSON")
+	}
+
+	c, _ := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/bulk_create",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+	c.Set("identity", &identity.XRHID{Identity: identityHeader})
+
+	err = BulkCreate(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var users []m.User
```

Comment:
similar here ... the test assumes that table users is empty, but it may be different ... of course we can change it in these tests after we add some conflicting tests in the future